plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -530,48 +529,71 @@ public boolean isCanceled() {
+								String url = OpenShiftServerUtils.getConnectionURL(server);
 								try {
-									String conUrl = server.getAttribute(OpenShiftServerUtils.ATTR_CONNECTIONURL, (String)null);
-									if( conUrl != null ) {
-										ConnectionURL conurl2 = ConnectionURL.forURL(conUrl);
-										IConnection con = ConnectionsRegistrySingleton.getInstance().getByUrl(conurl2);
+									if( url != null ) {
+										ConnectionURL conUrl = ConnectionURL.forURL(url);
+										IConnection con = ConnectionsRegistrySingleton.getInstance().getByUrl(conUrl);
 										model.loadResources(con);
 									}
-								} catch(UnsupportedEncodingException uee) {
-									
-								} catch(MalformedURLException murle) {
-									
+								} catch(UnsupportedEncodingException | MalformedURLException e) {
+									OpenShiftUIActivator.log(IStatus.ERROR, getConnectionErrorMessage(url, server));
 								}
 								return Status.OK_STATUS;
 							}
 						})
-				.runWhenDone(
-						new Job("Setting connection, deploy project...") {
+				.runWhenDoneIf(job -> {
+							if (connection == null) {
+								container.getDisplay().syncExec(() -> 
+									new ErrorDialog(container.getShell(), "Error", 
+											NLS.bind("Could not initialize server editor {0}", server.getName()),
+											OpenShiftUIActivator.statusFactory().errorStatus(
+													getConnectionErrorMessage(OpenShiftServerUtils.getConnectionURL(server), server)),
+											IStatus.ERROR)
+										.open());
+								return false;
+							} else {
+								return true;
+							}
+						}, new Job("Setting connection, deploy project...") {
 
 							@Override
 							protected IStatus run(IProgressMonitor monitor) {
+								if (connection == null) {
+									return Status.CANCEL_STATUS;
+								}
+
 								if (deployProject != null) {
 									model.setDeployProject(deployProject);
 								}
-								if (connection != null) {
-									model.setConnection(connection);
-								}
+								model.setConnection(connection);
 								model.setService(OpenShiftServerUtils.getService(server));
 								String sourcePath = OpenShiftServerUtils.getSourcePath(server);
-								if(!StringUtils.isEmpty(sourcePath)) {
+								if (!StringUtils.isEmpty(sourcePath)) {
 									model.setSourcePath(sourcePath);
 								}
 								String podPath = OpenShiftServerUtils.getPodPath(server);
 								if(!StringUtils.isEmpty(podPath)) {
 									model.setPodPath(podPath);
 								}
+
 								model.setInitializing(false);
 								return Status.OK_STATUS;
 							}
 						})
 				.runWhenDone(new DisableAllWidgetsJob(false, container, false, busyCursor))
 				.schedule();
 	}
+	
+	private String getConnectionErrorMessage(String url, IServerAttributes server) {
+		ConnectionURL connectionUrl = ConnectionURL.safeForURL(OpenShiftServerUtils.getConnectionURL(server));
+		if (connectionUrl == null) {
+			return "Could not find OpenShift connection for server \"{0}\"";
```

Comment:
It is very unlikely that we ever translate it to any other language than English. In the sake of efficiency I personally never extract the messages, I only (mostly) make sure that I bind messages (NLS.bind) so that extracting would be as simple as launching the internationalization wizard in Eclipse. Unfortunately there are peope  with different opinions and we were not strict enough to impose either way so that there are some messages and then there are lots on texts that are not extracted.